go/stellar/wallet_state.go

```diff
@@ -407,8 +410,12 @@ func (w *WalletState) SubmitRelayClaim(ctx context.Context, post stellar1.RelayC
 // MarkAsRead is an override of remoter's MarkAsRead.
 func (w *WalletState) MarkAsRead(ctx context.Context, accountID stellar1.AccountID, mostRecentID stellar1.TransactionID) error {
 	err := w.Remoter.MarkAsRead(ctx, accountID, mostRecentID)
-	mctx := libkb.NewMetaContext(ctx, w.G())
-	w.Refresh(mctx, accountID, "MarkAsRead")
+	if err == nil {
+		mctx := libkb.NewMetaContext(ctx, w.G())
+		if rerr := w.Refresh(mctx, accountID, "MarkAsRead"); rerr != nil {
```

Comment:
should this kind of thing by background/async? you'd know better whether this function returning is on any latency critical path

Reply:
good idea...i'll change it.